### PR TITLE
Welcome: Port Welcome to GML Compiler

### PR DESCRIPTION
**Pull request by mrunix00**
Contributes to: https://github.com/SerenityOS/serenity/issues/20518

### CODE REVIEW

File: Userland/Applications/Welcome/WelcomeWindow.gml [hunk 1]
```diff
@@ -25,7 +25,7 @@
                 min_width: 340
                 min_height: 170
                 fill_with_background_color: true
-                background_role: "Base"
+                background_role: @ColorRole::Base
```
Comment (AtkinsSJ) [inline Userland/Applications/Welcome/WelcomeWindow.gml:28]: This change (and the similar changes below) is incorrect. The problem is the GML compiler doesn't know that this property accepts an enum. If you add these properties to the HashMap here, and then revert them to the previous quoted strings, I think that should fix it:
https://github.com/SerenityOS/serenity/blob/5e1499d10450bb093035eadeb8e16cf5151ab047/Meta/Lagom/Tools/CodeGenerators/GMLCompiler/main.cpp#L173-L183

Comment (mrunix00) [reply Userland/Applications/Welcome/WelcomeWindow.gml]: Done!

**Review by kleinesfilmroellchen — APPROVED** [review]

File: Meta/Lagom/Tools/CodeGenerators/GMLCompiler/main.cpp [hunk 2]
```diff
@@ -174,12 +174,14 @@ static ErrorOr<Optional<String>> generate_enum_initializer_for(StringView proper
         { "text_alignment"sv, "Gfx::TextAlignment"sv },
         { "focus_policy"sv, "GUI::FocusPolicy"sv },
         { "foreground_role"sv, "Gfx::ColorRole"sv },
+        { "background_role"sv, "Gfx::ColorRole"sv },
```
Comment (timschumi) [inline Meta/Lagom/Tools/CodeGenerators/GMLCompiler/main.cpp:177]: This appears to have slipped in from the other PR.

File: Meta/Lagom/Tools/CodeGenerators/GMLCompiler/main.cpp [hunk 3]
```diff
@@ -174,12 +174,14 @@ static ErrorOr<Optional<String>> generate_enum_initializer_for(StringView proper
...
         { "frame_style"sv, "Gfx::FrameStyle"sv },
         { "text_wrapping"sv, "Gfx::TextWrapping"sv },
         { "button_style"sv, "Gfx::ButtonStyle"sv },
         { "opportunistic_resizee"sv, "GUI::Splitter::OpportunisticResizee"sv },
         { "checkbox_position"sv, "GUI::CheckBox::CheckBoxPosition"sv },
-        { "button_style"sv, "Gfx::ButtonStyle"sv }
+        { "button_style"sv, "Gfx::ButtonStyle"sv },
+        { "font_weight"sv, "Gfx::FontWeight"sv }
```
Comment (timschumi) [inline Meta/Lagom/Tools/CodeGenerators/GMLCompiler/main.cpp:184]: A trailing comma should be fine here.

**Review by timschumi — CHANGES_REQUESTED** [review]

File: Meta/Lagom/Tools/CodeGenerators/GMLCompiler/main.cpp (hunk 2 above)
Comment (mrunix00) [reply Meta/Lagom/Tools/CodeGenerators/GMLCompiler/main.cpp:177]: Oh yeah, I guess I'll change it when the other PR is accepted
Next time when I need to port multiple apps to the GML compiler I'll do it in a single PR

**Review by timschumi — APPROVED** [review]

**Pull request merged**